app/models/docker_builder_service.rb

```diff
@@ -132,8 +133,22 @@ def run_build_image_job(local_job, push: false, tag_as_latest: false)
     build.save!
   end
 
-  def build_image(tmp_dir)
+  def before_docker_build(tmp_dir)
+    before_docker_build_file = File.join(tmp_dir, BEFORE_DOCKER_BUILD)
```

Comment:
ideally just be another hook ... but afaik there is no nice way of setting this up without creating a super tiny plugin :/